tools/functional-tester/etcd-runner/main.go

```diff
@@ -310,10 +320,24 @@ func runWatcher(eps []string) {
 						return
 					}
 				}
-			}(prefixes[i])
+			}(roundPrefix + "-" + prefixes[i])
 		}
 	}
 	wg.Wait()
+
+	// cancel all watch channels
+	cancelc()
+
```

Comment:
@sinsharat clientv3.Close() was only tearing down the grpc connection, the watch client would notice the grpc close and then tear down the watchers asynchronously. This races with `cancelc()`; some channels would close through `cancelc` and others would close through the grpc connection close. As of #6636, clientv3.Close synchronously tears down the watchers.